src/ui/components/Table/types.ts

```diff
@@ -0,0 +1,68 @@
+import React from 'react';
+
+import {RowData} from '@tanstack/react-table';
+
+declare module '@tanstack/react-table' {
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    interface ColumnMeta<TData extends RowData, TValue> {
+        width?: string | number;
+        footer?: {
+            css?: React.CSSProperties;
+        };
+        head?: THead;
+    }
+}
+
+interface CellData {
+    value: unknown;
+    formattedValue?: string;
+    css?: React.CSSProperties;
+    className?: string | (() => string);
+
+    rowSpan?: number;
+    isVisible?: boolean;
+}
+
+export type RenderCellFn<T extends CellData> = (cellData: T) => React.ReactElement | null;
+export type RenderHeaderFn = () => React.ReactElement | null;
+
+export type THead = {
+    id: string;
+    header?: string | RenderHeaderFn;
+    enableSorting?: boolean;
+    enableRowGrouping?: boolean;
+    width?: string | number;
+    cell?: RenderCellFn<CellData>;
+    columns?: THead[];
+};
+
+export type TData = CellData[];
+
+export type TFoot = CellData;
+
+export type OnTableClick = (args: {row?: TData; cell?: CellData; event: React.MouseEvent}) => void;
+
+export type TableProps = {
+    title?: {text: string};
+    data: {
+        head?: THead[];
+        rows?: TData[];
+        footer?: TFoot[];
+    };
+    noData?: {
+        text: string;
+    };
+    header?: {
+        sticky?: boolean;
+    };
+    onClick?: OnTableClick;
+    qa?: {
```

Comment:
It looks complicated, besides all of such properties in gravity are strings and it will be inconsistent. I think it is quite enough to have one `qa` attribute on table itself